src/read/xcoff/section.rs

```diff
@@ -0,0 +1,380 @@
+};
+
+use crate::read::{self, ObjectSection, ReadError, ReadRef, Result, SectionIndex};
+
+use super::{FileHeader, XcoffFile, XcoffRelocationIterator};
+
+/// An iterator over the sections of an `XcoffFile32`.
+pub type XcoffSectionIterator32<'data, 'file, R = &'data [u8]> =
+    XcoffSectionIterator<'data, 'file, xcoff::FileHeader32, R>;
+/// An iterator over the sections of an `XcoffFile64`.
+pub type XcoffSectionIterator64<'data, 'file, R = &'data [u8]> =
+    XcoffSectionIterator<'data, 'file, xcoff::FileHeader64, R>;
+
+/// An iterator over the sections of an `XcoffFile`.
+#[derive(Debug)]
+pub struct XcoffSectionIterator<'data, 'file, Xcoff, R = &'data [u8]>
+where
+    Xcoff: FileHeader,
+    R: ReadRef<'data>,
+{
+    pub(super) file: &'file XcoffFile<'data, Xcoff, R>,
+    pub(super) iter: iter::Enumerate<slice::Iter<'data, Xcoff::SectionHeader>>,
+}
+
+impl<'data, 'file, Xcoff, R> Iterator for XcoffSectionIterator<'data, 'file, Xcoff, R>
+where
+    Xcoff: FileHeader,
+    R: ReadRef<'data>,
+{
+    type Item = XcoffSection<'data, 'file, Xcoff, R>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.iter.next().map(|(index, section)| XcoffSection {
+            index: SectionIndex(index),
+            file: self.file,
+            section,
+        })
+    }
+}
+
+/// A section of an `XcoffFile32`.
+pub type XcoffSection32<'data, 'file, R = &'data [u8]> =
+    XcoffSection<'data, 'file, xcoff::FileHeader32, R>;
+/// A section of an `XcoffFile64`.
+pub type XcoffSection64<'data, 'file, R = &'data [u8]> =
+    XcoffSection<'data, 'file, xcoff::FileHeader64, R>;
+
+/// A section of an `XcoffFile`.
+#[derive(Debug)]
+pub struct XcoffSection<'data, 'file, Xcoff, R = &'data [u8]>
+where
+    'data: 'file,
+    Xcoff: FileHeader,
+    R: ReadRef<'data>,
+{
+    pub(super) file: &'file XcoffFile<'data, Xcoff, R>,
+    pub(super) section: &'data Xcoff::SectionHeader,
+    pub(super) index: SectionIndex,
+}
+
+impl<'data, 'file, Xcoff: FileHeader, R: ReadRef<'data>> XcoffSection<'data, 'file, Xcoff, R> {
+    fn bytes(&self) -> Result<&'data [u8]> {
+        self.section
+            .xcoff_data(self.file.data)
+            .read_error("Invalid XCOFF section offset or size")
+    }
+}
+
+impl<'data, 'file, Xcoff, R> read::private::Sealed for XcoffSection<'data, 'file, Xcoff, R>
+where
+    Xcoff: FileHeader,
+    R: ReadRef<'data>,
+{
+}
+
+impl<'data, 'file, Xcoff, R> ObjectSection<'data> for XcoffSection<'data, 'file, Xcoff, R>
+where
+    Xcoff: FileHeader,
+    R: ReadRef<'data>,
+{
+    type RelocationIterator = XcoffRelocationIterator<'data, 'file, Xcoff, R>;
+
+    fn index(&self) -> SectionIndex {
+        self.index
+    }
+
+    fn address(&self) -> u64 {
+        self.section.s_paddr().into()
+    }
+
+    fn size(&self) -> u64 {
+        self.section.s_size().into()
+    }
+
+    fn align(&self) -> u64 {
+        // TODO: return the alignment for different sections.
+        4
```

Comment:
In XCOFF object file, the default section alignment is 4.
The `o_algntext` and `o_algndata` fields in auxiliary header can only be found in executable files.
This patch doesn't fully support the executable format, so I add a TODO comment for it.

Reply:
> In XCOFF object file, the default section alignment is 4.

Is this documented somewhere?

Is it a default section alignment (meaning it can be changed somewhere), or is it a fixed section alignment (meaning it is always 4)?

Is your TODO comment referring to `o_algntext` and `o_algndata` or something else?

Reply:
> In XCOFF object file, the default section alignment is 4.

This is a definition used in LLVM compiler https://github.com/llvm/llvm-project/blob/6dcd15aed121a428799b028b771711c01b47ef29/llvm/lib/MC/XCOFFObjectWriter.cpp#L51. I'm just following this definition.

> Is your TODO comment referring to o_algntext and o_algndata or something else?

Yes, I've updated the comments now.